internal/generator/kzg/template/mpcsetup.go.tmpl

```diff
@@ -113,12 +116,26 @@ func (s *MpcSetup) Verify(next *MpcSetup) error {
 	}
 
 	if !next.srs.Vk.G2[1].IsInSubGroup() {
-		return errors.New("𝔾₂ representation not in subgroup")
+		return errors.New("[x]₂ representation not in subgroup")
 	}
-	for i := 1; i < len(next.srs.Pk.G1); i++ {
-		if !next.srs.Pk.G1[i].IsInSubGroup() {
-			return errors.New("𝔾₁ representation not in subgroup")
+
+	// TODO @Tabaie replace with batch subgroup check
+	n := len(next.srs.Pk.G1) - 1
+	fail := -1
+	wp := utils.NewWorkerPool()
+	defer wp.Stop()
+
+	wp.Submit(n, func(start, end int) {
+		for i := start; i < end; i++ {
+			if !next.srs.Pk.G1[i+1].IsInSubGroup() {
+				fail = i + 1
```

Comment:
that does not work; fail can be accessed by multiple go routines .
Either use a chan of error that you check at the end, or at atomic counter that you increment